proposals/records-wip.md

```diff
@@ -89,13 +89,14 @@ A `with` expression allows for "non-destructive mutation", designed to
 produce a copy of the receiver expression with modifications to properties
 listed in the `anonymous_object_initializer`.
 
-A valid `with` expression has a receiver with a non-void type. The receiver type must contain an accessible parameterless instance method called `Clone` whose return type must be the type of the receiver express type, or a base type thereof.
+A valid `with` expression has a receiver with a non-void type. The receiver type must contain an accessible parameterless instance method called `Clone` whose return type must be the receiver type, or a base type thereof.
 
 On the right hand side of the `with` expression is an `anonymous_object_initializer` with a
-sequence of assignments with a compiler-generated record property of the receiver on the left-hand side of the
-assignment, and an arbitrary expression on the right-hand side which is implicitly convertible to the type
-of the left-hand side.
+sequence of assignments with a compiler-generated record property (see Properties, above) of the receiver 
+on the left-hand side of the assignment, and an arbitrary expression on the right-hand side which is 
+implicitly convertible to the type of the left-hand side.
 
 The evaluation of a `with` expression is equivalent to calling the `Clone` method exactly once,
 and then setting the backing field of each record property in the argument list to its corresponding
 expression, in lexical order, using the result of the `Clone` method as the receiver.
```

Comment:
>using the result of the `Clone` method as the receiver [](start = 30, length = 54)

The `Clone` method might not have that property as a member (in the case that it returns a base type).

Reply:
Fixed. The properties should in the return type of `Clone`, not the receiver type